src/main/java/org/jenkins/tools/test/hook/NodeCleanupBeforeCompileHook.java

```diff
@@ -19,16 +23,15 @@ public Map<String, Object> action(Map<String, Object> moreInfo) throws Exception
         if (shouldExecuteHook) {
             File pluginDir = (File) moreInfo.get("pluginDir");
             try {
-                System.out.println("Executing node and node_modules cleanup hook");
+                LOGGER.log(Level.INFO, "Executing node and node_modules cleanup hook");
                 compile(pluginDir);
                 return moreInfo;
             } catch (Exception e) {
-                System.out.println("Exception executing hook");
-                System.out.println(e);
+                LOGGER.log(Level.WARNING, "Exception executing hook", e);
```

Comment:
All this logging and throwing is itself an anti-pattern that could be cleaned up in a future PR that deals with error handling and propagation, but is explicitly out of scope for this PR.